node/src/actors/chain_manager/mod.rs

```diff
@@ -258,14 +262,28 @@ impl ChainManager {
         )
         .into_actor(self)
         .and_then(|_, _, _| {
-            log::trace!("Successfully persisted chain_info into storage");
+            log::debug!("Successfully persisted chain_info into storage");
             fut::ok(())
         })
         .map_err(|err, _, _| log::error!("Failed to persist chain_info into storage: {}", err))
         .wait(ctx);
     }
     /// Method to persist the chain_state into storage
     fn persist_chain_state(&mut self, ctx: &mut Context<Self>) {
+        // When persisting the chain state, we need to update the highest superblock checkpoint.
+        // This is the highest superblock that obtained a majority of votes and we do not want to
+        // lose it when restoring the state.
+        self.last_chain_state
+            .chain_info
+            .as_mut()
+            .unwrap()
+            .highest_superblock_checkpoint = self
+            .chain_state
+            .chain_info
+            .as_ref()
+            .unwrap()
+            .highest_superblock_checkpoint;
```

Comment:
there is a function called get_superblock_beacon() that returns exactly the same in line 688 in chain_managers/mod.rs